decompiler/util/commandline.py

```diff
@@ -71,16 +71,16 @@ def main(interface: "Decompiler"):
     try:
         if args.all or not args.function:
             # decompile all functions.
-            undecorated_code = decompiler.decompile_all(options)
-            DecoratedCode.print_code(
-                undecorated_code, output_stream, color, style=options.getstring("code-generator.style_cmd", fallback="paraiso-dark")
-            )
-        else:
-            for function_name in args.function:
-                task = decompiler.decompile(function_name, options)
+            if (undecorated_code := decompiler.decompile_all(options)):
                 DecoratedCode.print_code(
-                    task.code, output_stream, color, style=task.options.getstring("code-generator.style_cmd", fallback="paraiso-dark")
+                    undecorated_code, output_stream, color, style=options.getstring("code-generator.style_cmd", fallback="paraiso-dark")
                 )
+        else:
+            for function_name in args.function:
+                if (undecorated_code := decompiler.decompile(function_name, options)):
+                    DecoratedCode.print_code(
+                        undecorated_code, output_stream, color, style=options.getstring("code-generator.style_cmd", fallback="paraiso-dark")
+                    )
```

Comment:
I don't quiet get he point of this changes - care to ellaborate?

if function now consists of an if checking if `args.function` was set and it iterates over it, when it was not set... O_o

Reply:
Don't really get the comment here. 
I just changed the variable to reflect the type of  (str instead of DecompilerTask) decompile/decompile_all. 

The wrapper into the if expression is for the case, that the generated code is empty (which would be the case if one/all methods could not be loaded by bninja)